pkg/logql/log/labels.go

```diff
@@ -136,7 +136,7 @@ func (b *BaseLabelsBuilder) ForLabels(lbs labels.Labels, hash uint64) *LabelsBui
 }
 
 // Reset clears all current state for the builder.
-func (b *LabelsBuilder) Reset() {
+func (b *BaseLabelsBuilder) Reset() {
```

Comment:
LabelBuilders are BaseLabelBuilders so this change makes `Reset` available to `Pipeline` without affecting it's visibility in `StreamPipeline`